lms/djangoapps/instructor_task/tasks_helper/grade_utils.py

```diff
@@ -60,7 +60,7 @@ def calculate_students_grades_report(course_id):
 
             output.append(headers)
 
-        student_cohort = student.course_groups.first().name if student.course_groups.first() else '-'
+        student_cohort = student.course_groups.last().name if student.course_groups.last() else '-'
```

Comment:
Why you don't use CohortMembership here? As for it cleary provides garanty of the unique relation of the user-cohort.

Reply:
Done